tests/check_framework/test_caches.py

```diff
@@ -89,3 +91,24 @@ def test_cache_path_not_conflict(self):
             settings = self.get_settings(setting, root / 'cache', root / 'other')
             with self.subTest(setting=setting), self.settings(**settings):
                 self.assertEqual(check_cache_location_not_exposed(None), [])
+
+class CheckCacheAbsolutePath(SimpleTestCase):
+    warning_message = (
+        "Your '%s' LocalBasedCache LOCATION specifies a relative directory. "
+        "Use an absolute directory to make it independent of the working directory "
+        "of the Django process."
+    )
```

Comment:
`warning_message` is used in a single test, there is no need to add it to the class.